tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -400,20 +400,35 @@ void test_variables_math() noexcept {
                                       vec_expect[2], vec_expect[3]});
 }
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Variables.PrefixSemantics",
-                  "[DataStructures][Unit]") {
+template <typename VectorType>
+void test_variables_prefix_semantics() noexcept {
   using variable_type =
-      Variables<tmpl::list<VariablesTestTags_detail::vector,
-                           VariablesTestTags_detail::scalar,
-                           VariablesTestTags_detail::scalar2>>;
-  using prefix_variable_type = Variables<tmpl::list<
-      VariablesTestTags_detail::PrefixTag0<VariablesTestTags_detail::vector>,
-      VariablesTestTags_detail::PrefixTag0<VariablesTestTags_detail::scalar>,
-      VariablesTestTags_detail::PrefixTag0<VariablesTestTags_detail::scalar2>>>;
-  prefix_variable_type prefix_vars_0(4, 8.);
-  prefix_variable_type prefix_vars_1(4, 6.);
-  variable_type vars_0(prefix_vars_0);
-  variable_type vars_1(std::move(prefix_vars_1));
+      Variables<tmpl::list<VariablesTestTags_detail::vector_tag<VectorType>,
+                           VariablesTestTags_detail::scalar_tag<VectorType>,
+                           VariablesTestTags_detail::scalar2_tag<VectorType>>>;
+  using prefix_variable_type = Variables<
+      tmpl::list<VariablesTestTags_detail::PrefixTag0<
+                     VariablesTestTags_detail::vector_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag0<
+                     VariablesTestTags_detail::scalar_tag<VectorType>>,
+                 VariablesTestTags_detail::PrefixTag0<
+                     VariablesTestTags_detail::scalar2_tag<VectorType>>>>;
+
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<
+      typename tt::get_fundamental_type_t<typename VectorType::value_type>>
```

Comment:
In general, the whole point of `_t` is so you don't need to add `typename` in front. Please remove the extra `typename` everywhere you're using it.